lib/Serialization/ModuleFile.cpp

```diff
@@ -1332,6 +1333,21 @@ ModuleFile::ModuleFile(
           SearchPaths.push_back({blobData, isFramework, isSystem});
           break;
         }
+        case input_block::FILE_DEPENDENCY: {
+          if (extInfo) {
+            StringRef file = llvm::sys::path::filename(blobData);
+            StringRef name = llvm::sys::path::stem(file);
+            StringRef ext = llvm::sys::path::extension(file);
+            StringRef interface = file_types::getExtension(
+                file_types::TY_SwiftParseableInterfaceFile);
+            if (ext.endswith(interface) && name == info.name) {
+              assert(extInfo->getParseableInterface().empty() &&
+                     "found more than one .swiftinterface");
+              extInfo->setParseableInterface(blobData);
```

Comment:
Instead of parsing this back out of the dependencies, ParseableInterfaceModuleLoader should just set it up for us.

Reply:
Done, although the plumbing doesn't look too nice.